woocommerce/woocommerce
# Add SplitDropdown component

### All Submissions:






### Changes proposed in this Pull Request:

Add a component called `SplitDropdown` to `@woocommerce/components`. Under the hood this is a `ButtonGroup` with the first child as its first element and then a `Dropdown` encapsulating the rest of the children.

Children are expected but not required to be `Button`s. Two props (`variant` and `disabled`) on the `SplitDropdown` itself are applied to the first child and the dropdown toggle.



### How to test the changes in this Pull Request:

1. From the repo root directory run `npm run storybook`
2. Verify the behavior of the various SplitDropdown stories in Storybook
3. Edit `packages/js/components/src/split-dropdown/stories/index.tsx` if you want to try different configurations

### Other information:




#### Screenshots

All defaults:
<img width="169" alt="Screen Shot 2022-08-04 at 14 48 54" src="https://user-images.githubusercontent.com/890809/182940081-c4fa353a-0ea9-4478-a14f-6a41feafa4c1.png">

With a `variant` prop of `'secondary'`:
<img width="195" alt="Screen Shot 2022-08-04 at 14 49 22" src="https://user-images.githubusercontent.com/890809/182940142-b31a9dd3-d640-4b2f-b6c1-a84467e55bb3.png">

Some actions disabled (Individual child buttons have `disabled` prop):
<img width="171" alt="Screen Shot 2022-08-04 at 14 49 39" src="https://user-images.githubusercontent.com/890809/182940185-b230a799-0fee-4080-bf56-a56edbe9f570.png">

Fully disabled (The parent component has the `disabled` prop):
<img width="195" alt="Screen Shot 2022-08-04 at 14 51 22" src="https://user-images.githubusercontent.com/890809/182940452-01b3dd51-b199-4a21-b7ba-c74ca5c261ef.png">

Note that the outlines around all these buttons are not consistent with the design. We think this is because of an out-of-date @wordpress/components package, but we'll need to revisit it.

### FOR PR REVIEWER ONLY:
